[DOCS-1422][DOCS-1375] Document workspace templates

mdlinville opened the pull request:
[DOCS-1422] Document workspace templates
[DOCS-1375] Add private preview admonition

Ready for peer review, can be merged ahead of 0.69 release notes on 5/27, so I have removed the DO NOT MERGE.

Preview: https://docs-1422.docodile.pages.dev/guides/track/workspaces/#workspace-templates

[DOCS-1422]: https://wandb.atlassian.net/browse/DOCS-1422?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DOCS-1375]: https://wandb.atlassian.net/browse/DOCS-1375?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mdlinville committed:
[DOCS-1422] Document workspace templates

mdlinville commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,45 @@ Remove saved views that are no longer needed.
+| Group aggregation  | Mean |
+
+### Create a workspace template
+1. Open or create a workspace that has the [line plot settings]({{< relref "https://docs.wandb.ai/guides/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}) you want to templatize.
```
```suggestion
1. Open or create a workspace that has the [line plot settings]({{< relref "/guides/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}) you want to templatize.
```

mdlinville committed:
Update content/guides/models/track/workspaces.md

ktmathews89 commented:
I actually think it might be better to have @tristan-spaulding2 opinion on this docs update....

ktmathews89 commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,45 @@ Remove saved views that are no longer needed.
+    If you see **Save personal workspace template**, no workspace template has been created yet. Refer to [Create a workspace template]({{< relref "#create-a-workspace-template" >}}).
+    {{% /alert %}}
+1. Verify the new line plot settings, then click **Update**. The template is updated and reapplied to all workspaces that use it.
+
```
these all use `1.`

mdlinville replied:
Yep, if you look at the HTML preview you will see that it auto-numbers them correctly.

ktmathews89 commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,45 @@ Remove saved views that are no longer needed.
+### Update a workspace template
+After you create a workspace template, follow these steps to update it:
+
+1. Open any workspace that uses the existing template.
```
you can really just go to or open any workspace... you don't need to open one that uses and existing template

mdlinville replied:
This is confusing. How can I know if a workspace is currently using the template or not?

ktmathews89 replied:
hmmm why do you think that is important to the user?

you can change the settings or the workspace at anytime... this is really just trying to give you a better default experience.

mdlinville replied:
I think I got my mental map straightened out in the interim. Is there more I need to correct?

mdlinville committed:
Merge branch 'main' into DOCS-1422

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1422

mdlinville committed:
Katie's feedback

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1422' into DOCS-1422

tristan-spaulding2 commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,50 @@ Remove saved views that are no longer needed.
+
+{{% alert %}}
+To stop using a customized workspace template, you can [update your workspace template]({{< relref "#update-a-workspace-template" >}}) back to the [default setting]({{< relref "#default-workspace-settings" >}}).
+{{% /alert %}}
```
We probably can delete this since you also add this warning at the end, right?

tristan-spaulding2 commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,50 @@ Remove saved views that are no longer needed.
+1. To save the changes to the template, click the workspace's `...` action menu at the top of the page near the **Undo** and **Redo** arrow icons, then click **Update personal workspace template**.
+    {{% alert %}}
+    If you see **Save personal workspace template**, no workspace template has been created yet. Refer to [Configure your workspace template]({{< relref "#configure-your-workspace-template" >}}).
+    {{% /alert %}}
```
This felt a bit awkward, do we need it?

mdlinville replied:
I'll take it out. I was worried about confusion if the user doesn't see the exact text, but perhaps they will be able to figure it out.

tristan-spaulding2 commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,50 @@ Remove saved views that are no longer needed.
 ### Share a workspace view
 Share your customized workspace with your team by sharing the workspace URL directly. All users with access to the workspace project can see the saved Views of that workspace.
 
+## Workspace templates
```
We likely need a callout letting people know this is only available for Enterprise Edition customers.

mdlinville replied:
Added this to the same admonition where the private preview detail is shown. When we remove that, I'll shift this to the normal enterprise-only admonition.

mdlinville replied:
Updated this after Slack conversation to note that it is Enterprise-only for everyone, and private preview for Dedicated Cloud and Self-Hosted.

mdlinville committed:
Merge branch 'main' into DOCS-1422

ktmathews89 reviewed (approved):
the changes look great!

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1422

mdlinville committed:
[DOCS-1375] Add private preview admonition

mdlinville committed:
Use private preview admonition for workspaces

mdlinville committed:
Tristan's feedback

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1422' into DOCS-1422

mdlinville committed:
Merge branch 'main' into DOCS-1422

ngrayluna commented on content/guides/models/app/features/panels/line-plot/_index.md:
```diff
@@ -1,12 +1,12 @@
 ---
 description: Visualize metrics, customize axes, and compare multiple lines on a plot
-url: guides/app/features/panels/line-plot
+url: /guides/app/features/panels/line-plot
```
Should we add "models" to the url path?

mdlinville replied:
The URL is https://docs.wandb.ai/guides/app/features/panels/line-plot/. (Verified with the public docs). I thought that the leading slash was needed for some reason so I added it here, but it looks like none of the others have it so I will revert this. It doesn't seem to matter either way.

ngrayluna commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,50 @@ Remove saved views that are no longer needed.
+This feature requires an [Enterprise](https://wandb.ai/site/pricing/) license.
+{{% /alert %}}
+
+Use _workspace templates_ to quickly create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
```
```suggestion
Use _workspace templates_ to create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
```

ngrayluna replied:
"Default workspace settings" make it sound like there are multiple settings that can be configured (e.g. layout of the workspace, how many panels show up, etc.). 

Let's be more specific and mentions that it only (or seems to only)  apply to line plot panels.

mdlinville replied:
@ktmathews89 @tristan-spaulding2 WDYT? I left Katie's language in place to let the reader know that more settings are coming. If I do the change Noah is suggesting, I worry that the reader will think that the only thing we ever plan to include is line plot settings.

ngrayluna replied:
With the update, keep in mind not dating the docs: https://developers.google.com/style/timeless-documentation

ktmathews89 replied:
it's going to be expanded for additional settings... so I personally would keep it generalized.

ngrayluna commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,50 @@ Remove saved views that are no longer needed.
+
+1. Open any workspace.
+1. Modify the workspace's line plot settings. For example, set the number of runs to include to `11`.
+1. To save the changes to the template, click the workspace's `...` action menu at the top of the page near the **Undo** and **Redo** arrow icons, then click **Update personal workspace template**.
```
Do users know that "..." is known as an action menu? If not, should we describe what it looks like instead?

mdlinville replied:
Maybe the reader of _this_ page doesn't know that. In the panels pages, they do know. I'll adjust this a bit.

ngrayluna reviewed (changes requested):
Some minor comments

mdlinville committed:
Most of Noah's feedback

mdlinville committed:
Merge remote-tracking branch 'origin/main' into DOCS-1422

mdlinville committed:
Merge remote-tracking branch 'origin/DOCS-1422' into DOCS-1422

mdlinville committed:
Add more specific private preview include

- Enterprise license is required for the whole feature
- Dedicated and Self-Hosted: Private preview, enrollment required

tristan-spaulding2 reviewed (approved):
lgtm, thanks.

mdlinville commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,49 @@ Remove saved views that are no longer needed.
+Use _workspace templates_ to quickly create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
+
+### Default workspace settings
+By default, new workspaces use these default settings for line plots:
```
```suggestion
By default, new workspaces use these default settings:

#### Line plots

```
Here's a compromise. :)

mdlinville commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,49 @@ Remove saved views that are no longer needed.
+{{< readfile file="/_includes/workspace-templates-private-preview-feature.md" >}}
+{{% /alert %}}
+
+Use _workspace templates_ to quickly create workspaces using the same settings as an existing workspace instead of the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}). Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
```
```suggestion
Use _workspace templates_ to automatically apply custom settings from an existing workspace to new workspaces without updating each [default setting]({{< relref "#default-workspace-settings" >}}) individually. Currently, a workspace template can define custom [line plot settings]({{< relref "/guides/models/app/features/panels/line-plot/#all-line-plots-in-a-workspace" >}}).
```

mdlinville committed:
Apply suggestions from code review

ngrayluna commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,52 @@ Remove saved views that are no longer needed.
+| Setting | Default |
+|-------|----------
+| X axis             | Step |
+| Smoothing type     | Time weight EMA |
```
Clarify that EMA == exponential moving average
```suggestion
| Smoothing type     | Time weight exponential moving average (EMA) |
```

ngrayluna commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,52 @@ Remove saved views that are no longer needed.
+1. To save the changes to the template, click the action menu `...` near the **Undo** and **Redo** arrow icons, then click **Update personal workspace template**.
+1. Verify the new line plot settings, then click **Update**. The template is updated, and reapplied to all workspaces that use it.
+
+{{% alert %}}
```
I wonder if it makes sense to have this info also (or instead) in the previous section? Just a thought.

mdlinville replied:
Done, also demoted this one from being an alert.

ngrayluna reviewed (commented):
Quick comments

mdlinville commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,52 @@ Remove saved views that are no longer needed.
+    1. Click **Save personal workspace template**.
+    1. Review the line plot settings for the template, then click **Save**.
+
+New workspaces will use these settings instead of the defaults.
```
```suggestion
New workspaces will use these settings instead of the defaults.

If you decide to stop using a customized workspace template, you can [update it]({{< relref "#update-your-workspace-template" >}}) to use the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}).
```

mdlinville commented on content/guides/models/track/workspaces.md:
```diff
@@ -61,6 +61,52 @@ Remove saved views that are no longer needed.
+
+{{% alert %}}
+If you decide to stop using a customized workspace template, you can [update it]({{< relref "#update-your-workspace-template" >}}) to use the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}).
+{{% /alert %}}
```
```suggestion
If you decide to stop using a customized workspace template, you can [update it]({{< relref "#update-your-workspace-template" >}}) to use the [default settings for new workspaces]({{< relref "#default-workspace-settings" >}}).
```

mdlinville committed:
Noah's feedback

Co-authored-by: Noah Luna <15202580+ngrayluna@users.noreply.github.com>

mdlinville committed:
Merge branch 'main' into DOCS-1422
